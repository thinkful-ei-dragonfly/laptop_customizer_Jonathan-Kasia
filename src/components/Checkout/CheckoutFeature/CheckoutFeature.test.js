import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import CheckoutFeature from './CheckoutFeature'

const selected = {
  Processor: {
      name: 'MacBookPro',
      cost: 700
    },
  "Operating System": {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    },
  "Video Card":{
      name: 'NVidia',
      cost: 1150.98
    },
  Display: {
      name: 'Sony',
      cost: 1500
    }
}

describe('CheckoutFeature Component', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<CheckoutFeature selected={selected} />, div)
    ReactDOM.unmountComponentAtNode(div) 
  })

  it('Renders the UI as expected', () => {
    const tree = renderer
    .create(<CheckoutFeature selected={selected}  />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  })
})