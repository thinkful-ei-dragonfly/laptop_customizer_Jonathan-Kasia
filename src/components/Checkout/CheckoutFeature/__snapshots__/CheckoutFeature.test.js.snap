// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutFeature Component Renders the UI as expected 1`] = `
Array [
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Processor
        
    </div>
    <div
      className="summary__option__value"
    >
      MacBookPro
    </div>
    <div
      className="summary__option__cost"
    >
      $700.00
    </div>
  </div>,
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Operating System
        
    </div>
    <div
      className="summary__option__value"
    >
      Ubuntu Linux 16.04
    </div>
    <div
      className="summary__option__cost"
    >
      $200.00
    </div>
  </div>,
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Video Card
        
    </div>
    <div
      className="summary__option__value"
    >
      NVidia
    </div>
    <div
      className="summary__option__cost"
    >
      $1,150.98
    </div>
  </div>,
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Display
        
    </div>
    <div
      className="summary__option__value"
    >
      Sony
    </div>
    <div
      className="summary__option__cost"
    >
      $1,500.00
    </div>
  </div>,
]
`;
